OpenRA.Game/Traits/Player/PlayerResources.cs

```diff
@@ -79,8 +79,18 @@ public bool TakeResources(int num)
 
 		public void GiveCash(int num)
 		{
-			Cash += num;
-			Earned += num;
+			checked
+			{
+				try
+				{
+					Cash += num;
+					Earned += num;
+				}
+				catch (OverflowException)
+				{
+					Log.Write("debug", "Player cash overflow cought.");
```

Comment:
I don't think this log is necessary, but either way `cought` is spelled incorrectly.
It should also set each value to int.MaxValue if there is an overflow.
I also think the checked block would be better inside the try block.